board/xilinx/zynqmp/zynqmp.c

```diff
@@ -344,22 +344,68 @@ int board_early_init_f(void)
 	return 0;
 }
 
-static int multi_boot(void)
+static u32 multi_boot_get(void)
 {
 	u32 multiboot;
 
-	multiboot = readl(&csu_base->multi_boot);
+	zynqmp_mmio_read(&csu_base->multi_boot, &multiboot);
 
-	printf("Multiboot:\t%d\n", multiboot);
+	return multiboot;
+}
+
+static u32 multi_boot_set(u32 multiboot)
+{
+	zynqmp_mmio_write(&csu_base->multi_boot, 0xFFFFFFFF, multiboot);
 
 	return 0;
 }
 
+#ifndef CONFIG_SPL_BUILD
+static int do_multi_boot(struct cmd_tbl *cmdtp, int flag,
+			 int argc, char * const argv[])
+{
+	int ret;
+	u32 multiboot;
+
+	/* If we just want to retrieve current value */
+	if (argc == 1) {
+		multiboot = multi_boot_get();
+
+		printf("Multiboot:\t%d\n", multiboot);
```

Comment:
Could you use hex format for all outputs of multiboot value? I think it would be more natural and readable.

Reply:
As we discussed below in the comments, the offset defines 32KB blocks, so it's better to print it as decimal value